lib/apiservers/engine/backends/container.go

```diff
@@ -1248,10 +1248,30 @@ func validateCreateConfig(config *types.ContainerCreateConfig) error {
 
 	// validate port bindings
 	if config.HostConfig != nil {
+		var ips []string
+		if addrs, err := clientIPv4Addrs(); err != nil {
+			log.Warnf("could not get address for client interface: %s", err)
+		} else {
+			ips = make([]string, len(addrs))
+			for i := range addrs {
+				ips[i] = addrs[i].IP.String()
+			}
+		}
+
 		for _, pbs := range config.HostConfig.PortBindings {
```

Comment:
Are you concerned about the new for loop I added? There should really be 1-2 addresses in that array.

The other nested loops are just an artifact of how things are presented to us by docker.